dhis_android_analytics/src/main/java/dhis2/org/analytics/charts/ui/GroupAnalyticsViewModel.kt

```diff
@@ -73,22 +82,27 @@ class GroupAnalyticsViewModel(
 
     fun fetchAnalytics(groupUid: String?) {
         currentGroup = groupUid
-        _analytics.value = when (mode) {
-            AnalyticMode.ENROLLMENT -> uid?.let {
-                charts.geEnrollmentCharts(uid)
-                    .map { ChartModel(it) }
-            } ?: emptyList()
-            AnalyticMode.PROGRAM -> uid?.let {
-                charts.getProgramVisualizations(groupUid, uid)
-                    .map { ChartModel(it) }
-            } ?: emptyList()
-            AnalyticMode.HOME ->
-                charts.getHomeVisualizations(groupUid)
-                    .map { ChartModel(it) }
-            AnalyticMode.DATASET -> uid?.let {
-                charts.getDataSetVisualizations(groupUid, uid)
-                    .map { ChartModel(it) }
-            } ?: emptyList()
+        viewModelScope.launch {
+            val result = async(context = Dispatchers.IO) {
+                when (mode) {
+                    AnalyticMode.ENROLLMENT -> uid?.let {
+                        charts.geEnrollmentCharts(uid)
+                            .map { ChartModel(it) }
+                    } ?: emptyList()
+                    AnalyticMode.PROGRAM -> uid?.let {
+                        charts.getProgramVisualizations(groupUid, uid)
+                            .map { ChartModel(it) }
+                    } ?: emptyList()
+                    AnalyticMode.HOME ->
+                        charts.getHomeVisualizations(groupUid)
+                            .map { ChartModel(it) }
+                    AnalyticMode.DATASET -> uid?.let {
+                        charts.getDataSetVisualizations(groupUid, uid)
+                            .map { ChartModel(it) }
+                    } ?: emptyList()
+                }
+            }
+            _analytics.value = result.await()
```

Comment:
we need to capture error @Balcan this could crash

Reply:
done!